httpx streaming via `iter_raw` raises `httpx.StreamConsumed`

Here's a pytest-based MRE (Minimal Reproducible Example) that's directly inspired by the [existing `test_streaming` httpx unit test](https://github.com/h2non/pook/blob/ab8e8eef07daa9c40f512c9ea547faad12b87072/tests/unit/interceptors/httpx_test.py#L47-L55):

```python
import pook
import httpx


URL = "https://httpbin.org/status/404"


@pook.activate
def _check_httpx_streaming_via(response_method):
    streamed_response = b"streamed response"
    pook.get(URL).times(1).reply(200).body(streamed_response).mock

    with httpx.stream("GET", URL) as r:
        read_bytes = list(getattr(r, response_method)(chunk_size=1))

    assert len(read_bytes) == len(streamed_response)
    assert bytes().join(read_bytes) == streamed_response


def test_httpx_streaming_via_iter_bytes():
    _check_httpx_streaming_via("iter_bytes")


def test_httpx_streaming_via_iter_raw():
    _check_httpx_streaming_via("iter_raw")

```

And here are its test results, using the latest published versions of httpx, pook and pytest at the moment of this writing:

```shell
$ mktmpenv 
$ pip install httpx pook pytest
$ pytest --verbose mre.py
========================================================== test session starts ===========================================================
platform linux -- Python 3.10.12, pytest-8.0.0, pluggy-1.4.0 -- /home/petar/.virtualenvs/tmp-10e08f6fbb6325b/bin/python
cachedir: .pytest_cache
rootdir: /home/petar/projects/open-source/pook
configfile: pyproject.toml
plugins: anyio-4.2.0
collected 2 items                                                                                                                        

mre.py::test_httpx_streaming_via_iter_bytes PASSED                                                                                 [ 50%]
mre.py::test_httpx_streaming_via_iter_raw FAILED                                                                                   [100%]

================================================================ FAILURES ================================================================
___________________________________________________ test_httpx_streaming_via_iter_raw ____________________________________________________

    def test_httpx_streaming_via_iter_raw():
>       _check_httpx_streaming_via("iter_raw")

mre.py:25: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../.virtualenvs/tmp-10e08f6fbb6325b/lib/python3.10/site-packages/pook/api.py:139: in wrapper
    fn(*args, **kw)
mre.py:14: in _check_httpx_streaming_via
    read_bytes = list(getattr(r, response_method)(chunk_size=1))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Response [200 OK]>, chunk_size = 1

    def iter_raw(
        self, chunk_size: typing.Optional[int] = None
    ) -> typing.Iterator[bytes]:
        """
        A byte-iterator over the raw response content.
        """
        if self.is_stream_consumed:
>           raise StreamConsumed()
E           httpx.StreamConsumed: Attempted to read or stream some content, but the content has already been streamed. For requests, this could be due to passing a generator as request content, and then receiving a redirect response or a secondary request as part of an authentication flow.For responses, this could be due to attempting to stream the response content more than once.

../../../.virtualenvs/tmp-10e08f6fbb6325b/lib/python3.10/site-packages/httpx/_models.py:876: StreamConsumed
======================================================== short test summary info =========================================================
FAILED mre.py::test_httpx_streaming_via_iter_raw - httpx.StreamConsumed: Attempted to read or stream some content, but the content has already been streamed. For requests, this could b...
====================================================== 1 failed, 1 passed in 0.22s =======================================================
```

--- a/src/pook/interceptors/_httpx.py
+++ b/src/pook/interceptors/_httpx.py
@@ -75,6 +75,12 @@ class MockedTransport(httpx.BaseTransport):
             request=httpx_request,
         )
 
+        # Allow to read the response on client side
+        res.is_stream_consumed = False
+        res.is_closed = False
+        if hasattr(res, "_content"):
+            del res._content
+
         return res
 
 

--- a/tests/unit/interceptors/httpx_test.py
+++ b/tests/unit/interceptors/httpx_test.py
@@ -44,15 +44,23 @@ def test_json():
     assert response.json() == {"title": "123abc title"}
 
 
-def test_streaming():
+def _check_streaming_via(response_method):
     streamed_response = b"streamed response"
     pook.get(URL).times(1).reply(200).body(streamed_response).mock
 
     with httpx.stream("GET", URL) as r:
-        read_bytes = list(r.iter_bytes(chunk_size=1))
+        read_bytes = list(getattr(r, response_method)(chunk_size=1))
 
     assert len(read_bytes) == len(streamed_response)
     assert bytes().join(read_bytes) == streamed_response
+
+
+def test_streaming_via_iter_bytes():
+    _check_streaming_via("iter_bytes")
+
+
+def test_streaming_via_iter_raw():
+    _check_streaming_via("iter_raw")
 
 
 def test_redirect_following():

